unittests/blocks_log_replay_tests.cpp

```diff
@@ -0,0 +1,152 @@
+#include <eosio/testing/tester.hpp>
+#include <boost/test/unit_test.hpp>
+
+// Test scenarios
+//    * replay through blocks log and irreverible blocks
+//    * replay stopping in the middle of blocks log and resuming
+//    * replay stopping in the middle of irreverible blocks and resuming
+//
+
+BOOST_AUTO_TEST_SUITE(blocks_log_replay_tests)
+
+using namespace eosio::testing;
+using namespace eosio::chain;
+
+struct blog_replay_fixture {
+   eosio::testing::tester chain;
+   uint32_t               last_head_block_num {0}; // head_block_num at stopping
+   uint32_t               last_irreversible_block_num {0}; // LIB at stopping
+
+   // Activate Savanna and create blocks log
+   blog_replay_fixture() {
+      // Activate Savanna
+      size_t num_keys    = 21u;
+      size_t finset_size = 21u;
```

Comment:
I think 21 may be overkill, 4 is probably fine.

Reply:
I was thinking about simulating the actual network. But will change to 4 to save testing time.